roles/barbican/tasks/main.yml

```diff
@@ -0,0 +1,142 @@
+---
+- name: Check if barbican user exists
+  action: shell getent passwd barbican
+  register: barbican_user
+  failed_when: False
+  changed_when: False
+
+- name: barbican user
+  user:
+    name: barbican
+    shell: /bin/false
+    createhome: no
+  when: barbican_user|success
+
+- name: barbican user
+  user:
+    name: barbican
+    comment: barbican
+    shell: /bin/false
+    system: yes
+    home: /nonexistent
+    createhome: no
+  when: barbican_user|failed
+
+- name: /etc/barbican
+  file:
+    dest: /etc/barbican
+    state: directory
+    owner: barbican
+    group: barbican
+
+- name: barbican log dir
+  file:
+    dest: /var/log/barbican
+    state: directory
+    mode: 755
+    owner: barbican
+    group: adm
+
+- name: Change barbican log dir acl
+  acl: name=/var/log/barbican state=present default=yes etype={{ item.etype }} permissions={{ item.permission }}
+  with_items:
+    - etype: user
+      permission: rw
+    - etype: group
+      permission: r
+    - etype: other
+      permission: r
+
+- name: create barbican cache dir
+  file:
+    dest: /var/cache/barbican
+    state: directory
+    mode: 0700
+    owner: barbican
+    group: barbican
+
+- name: set uwsgi path (source install)
+  set_fact: uwsgi_path={{ openstack_source.virtualenv_base }}/barbican/bin/uwsgi
+  when: openstack_install_method == 'source'
+
+- name: set uwsgi path (package install ubuntu)
+  set_fact:
+    uwsgi_path: "{{ openstack_package.virtualenv_base }}/barbican/bin/uwsgi"
+  when: openstack_install_method == 'package' and ursula_os == 'ubuntu'
+
+- name: install barbican uwsgi service (ubuntu)
+  template:
+    src: etc/init/barbican.conf
+    dest: /etc/init/barbican.conf
+    mode: 0644
+  when: ursula_os == 'ubuntu'
+
+- name: Distribute Barbican Config Files
+  template:
+    src: "{{ item }}"
+    dest: /etc/barbican
+    mode: 0644
+    owner: barbican
+    group: barbican
+  with_fileglob: ../templates/etc/barbican/*
+
+- name: Creates barbican uwsgi and httpd directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0775
+    owner: barbican
+    group: barbican
+    mode: 0775
+  with_items:
+    - /etc/barbican/uwsgi
+    - /run/uwsgi
```

Comment:
see PR 2733 on how uwsgi is namespaced so that keystone, nova placements and now barbican all can use it.